Create 2019-12-19-countdown-to-the-new-year-ember-template-lint.md (#449)
ember-learn/ember-blog: source/2019-12-19-countdown-to-the-new-year-ember-template-lint.md

---
title: Countdown to The New Year - Ember Template Lint
author: Melanie Sumner
tags: 2019, Accessibility, Addons, DecEmber
alias: 2019-12-19-countdown-to-the-new-year-ember-template-lint.md
responsive: true
---

**This is the 19th in our DecEmber series–"Countdown to The New Year: 31 Days of Ember Addons". We plan to highlight a new addon each day until the new year, and we hope you'll join us for the fun!**

## Day 19

> Today we are going to look at an addon that, while it is built into Ember apps by default, has had enough updates lately to deserve a closer look- [ember-template-lint](https://github.com/ember-template-lint/ember-template-lint).

Reviewer: ```suggestion
Today we are going to look at an addon that is part of every new Ember app. The [ember-template-lint](https://github.com/ember-template-lint/ember-template-lint) addon has many new features added recently, and is worth a close look!
```

Rewording because the sentence seemed too complex

READMORE

### What It Does

> As you may guess from the addon title, `ember-template-lint` is a linter for your template files! 

Reviewer: ```suggestion
As you may guess from the addon title, `ember-template-lint` is a linter for your template files!
It checks for mistakes and advises best practices for the markdown in your component and route templates.
```

> Because we work in JavaScript, there are different ways something *could* be done- but often there is an ideal solution that we know won't cause problems down the line. 

Reviewer: ```suggestion
Because we work in JavaScript, there are different ways something *could* be done, but often there is an ideal solution that avoids problems down the line. 
```

> This is where `ember-template-lint` shines- it will throw warnings and/or errors when you try to break one of these rules.

Reviewer: ```suggestion
This is where `ember-template-lint` shines. It will throw warnings and/or errors when you try to break one of these rules.
```

It's extendable and customizable too, so you can get the right combination of rules for *your* project.

### Why I Like It

> The `ember-template-lint` addon isn't actually an Ember addon (which is why it's not on [Ember Observer](https://emberobserver.com/)) but it *is* an essential addon that helps provide accuracy, consistency and best practices in your application. 

Reviewer: ```suggestion
The `ember-template-lint` addon isn't actually an Ember addon, in terms of its file structure (which is why it's not on [Ember Observer](https://emberobserver.com/)) but it *is* an essential ingredient that helps provide accuracy, consistency and best practices in your application. 
```

Recently, I have been contributing to it myself, adding in extra ways that will help Ember developers avoid accessibility pitfalls.

> Through that work, I have come to deeply appreciate this addon- the [linter's contributors](https://github.com/ember-template-lint/ember-template-lint/graphs/contributors) have really done an incredible job at helping all of us write better template code.

Reviewer: ```suggestion
Through that work, I have come to deeply appreciate this addon. The [linter's contributors](https://github.com/ember-template-lint/ember-template-lint/graphs/contributors) have really done an incredible job at helping all of us write better template code.
```
